tp1/freeway/Assets/Scripts/chicken.cs

```diff
@@ -27,13 +30,25 @@ public class chicken : MonoBehaviour {
 
 	private void checkInput() {
 		Vector3 currentPosition = transform.position;
-		if (Input.GetKey(KeyCode.UpArrow)) {
-			transform.position = new Vector3 (currentPosition.x, currentPosition.y + speed, currentPosition.z);			
-		} else if (Input.GetKey(KeyCode.DownArrow)) {
-			if (currentPosition.y - speed < 1f * assetsSize) {
-				transform.position = initialPosition;
-			} else {
-				transform.position = new Vector3 (currentPosition.x, currentPosition.y - speed, currentPosition.z);
+		if(this.tag.Equals ("Chicken")) {
+			if (Input.GetKey(KeyCode.W)) {
```

Comment:
Here I think is the same. The difference is that I'm not sure about how to solve it. I understood that we can set which input to use in any prefab. If we could abstract these behaviors  we could add as many chicken as we want.